JFunc/src/main/java/com/jfunc/validator/FunctionValidator.java

```diff
@@ -25,92 +35,197 @@
-        if (VoidTypeUtil.isVoid(methodMetaData)) {
-            objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-            objectNode.set(JfuncConstants.REASONS, VoidTypeUtil.getReasons(methodMetaData));
-            return objectNode;
-        }
-
-        // Check for any "Print" and "Log" statements in a method
-        if (!(skipPrintStatements && skipLogStatements)) {
-            ArrayNode reasonObjectNode = mapper.createArrayNode();
-            boolean isFunctionSet = false;
-            List<InternalFeild> internalFields = methodMetaData.getInternallyRefferedFields();
-            for (InternalFeild internalField : internalFields) {
-                if (!skipPrintStatements && FieldUtil.containsPrintStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForPrintStatements(internalField));
-                }
-                if (!skipLogStatements && FieldUtil.containsLogStatements(internalField)) {
-                    if (!isFunctionSet) {
-                        objectNode.put(JfuncConstants.ISFUNCTIONAL, false);
-                        isFunctionSet = true;
-                    }
-                    reasonObjectNode.add(FieldUtil.getReasonsForLogStatements(internalField));
-                }
-            }
-            if (!isFunctionSet) {
-                objectNode.put(JfuncConstants.ISFUNCTIONAL, true);
-            } else {
-                objectNode.set(JfuncConstants.REASONS, reasonObjectNode);
-            }
-        }
-        return objectNode;
-    }
+	private final String filePath;
+	private final Collection<File> classFiles;
+	private ClassReader classReader = null;
+	private ClassNode classNode;
+	private final List<FieldNode> fieldNodes = new ArrayList<FieldNode>();
+	private final AppMetadata appNode = new AppMetadata();
 
+	private List<ObjectNode> results = new ArrayList<ObjectNode>();
+
+	private final boolean skipLogStatements = false;
+	private final boolean skipPrintStatements = false;
+
+	public FunctionValidator(String path) throws FileNotFoundException {
+		this.filePath = path;
+		classFiles = new ArrayList<File>();
+		File file = new File(filePath);
+		if (file.isDirectory()) {
+			classFiles.addAll(FileUtils.listFiles(file, JfuncConstants.EXTENSIONS, true));
+		} else if (FileUtil.isFileExists(file)) {
+			classFiles.add(file);
+		} else {
+			throw new FileNotFoundException();
+		}
+
+		for (File classFile : classFiles) {
+			init(classFile);
+		}
+	}
+
+	private void init(File classFile) {
+
+		try (InputStream fileStream = new FileInputStream(classFile)) {
+			classReader = new ClassReader(fileStream);
+		} catch (IOException e) {
+			e.printStackTrace();
+		}
+		classNode = new ClassNode();
+		classReader.accept(classNode, ClassReader.EXPAND_FRAMES);
```

Comment:
Do we need ClassReader.EXPAND_FRAMES, documentation says it degrades the performance very badly...